### PR TITLE
Add the ValidateCodeActionModal step to the 2FA

### PR DESCRIPTION
**Pull request by shahinyan11**
### Fixed Issues

$ https://github.com/Expensify/App/issues/51166
PROPOSAL: https://github.com/Expensify/App/issues/51166#issuecomment-2427102709

### Tests

1. Go to expensify.com > sign-up > choose "1-9" to be redirected to NewDot
2. Complete the onboarding modal steps to have a workspace created.
3. Go to Settings > Security > Two-factor authentication
4. Verify that there isn't a "Hold up! We need you to..." error message and the `ValidateCodeActionModal` is displayed

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as in the Tests section.
### QA Steps

Same as in the Tests section.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/5d8ea676-9fbc-4bce-ac04-48a8337c2674

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/c3897543-5a17-4d53-af50-f1578922b7b2

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/de4d6d00-8e65-4853-a8fb-5809a08a2b43

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/899a0878-c0dd-41f4-894a-c2fc9770704f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/0ff2b051-1e42-49be-a762-b46dcefd860b

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/92cf6d31-6fe6-4451-af9e-06f82cd2cd9a

</details>

### CODE REVIEW

**Comment by hungvu193** [comment]
Let's merge https://github.com/Expensify/App/pull/51856 into this one please.

**Comment by shahinyan11** [comment]
> Let's merge #51856 into this one please.

I moved the changes from https://github.com/Expensify/App/pull/51856here and closed it.

**Review by hungvu193 — COMMENTED** [review]
I have a small concern on the UI change, otherwise this is looking good!

File: src/components/ValidateCodeActionModal/index.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ function ValidateCodeActionModal({
                     onBackButtonPress={hide}
                 />
 
-                <View style={[themeStyles.ph5, themeStyles.mt3, themeStyles.mb7]}>
+                <View style={[themeStyles.ph5, themeStyles.mt3, themeStyles.mb7, themeStyles.flex1]}>
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/components/ValidateCodeActionModal/index.tsx:70]: Yeah this is definitely not a positioning we would want in this case cc @Expensify/design do you have a mock by any chance on how this should look?

[the remark this answers is not in the document]
Comment (shawnborton) [reply src/components/ValidateCodeActionModal/index.tsx:70]: Hmm why is that text even on this screen in the first place though? That should only show up on a screen like this:
CleanShot 2024-11-06 at 10 58 40@2x

**Comment by shawnborton** [comment]
Also, can we get some updated screenshots? It seems like the button is still not in the footer? 
CleanShot 2024-11-06 at 10 59 18@2x

**Comment by hungvu193** [comment]
We merged https://github.com/Expensify/App/pull/51856 into this one, I think @shahinyan11 forgot to update the screenshots.
Btw, you can see the updated screenshot here https://github.com/Expensify/App/pull/51856

**Comment by shawnborton** [comment]
Okay cool, it would still be good to update the screenshots here though for easier review!

**Comment by shahinyan11** [comment]
Should I update all screenshots/videos or will it be enough for one platform?

**Comment by shawnborton** [comment]
I think typically the screenshots need to be updated for all platforms and reflect the exact changes that this PR will introduce.

**Comment by shahinyan11** [comment]
I have removed the line mentioned in this [comment](https://github.com/Expensify/App/pull/51882#discussion_r1828657523) text and updated screenshots

**Comment by shahinyan11** [comment]
@hungvu193 @mountiny @shawnborton  In my option should that we have one more place where the `ValidateCodeActionModal`  should be added. What do you think ?

Screenshot 2024-11-06 at 15 35 09

**Comment by shawnborton** [comment]
Thanks! Screenshots look good to me 👍

**Review by hungvu193 — APPROVED** [review]
LGTM!

**Review by hungvu193 — APPROVED** [review]

**Review by mountiny — APPROVED** [review]
LGTM, thanks everyone!

**Comment by pecanoro** [comment]
@mountiny @shahinyan11 I think this PR caused this small-ish bug here: https://github.com/Expensify/App/issues/52336

**Comment by deetergp** [comment]
I also think it might have lead to this bug here https://github.com/Expensify/App/issues/52339

**Pull request merged**